<?xml version="1.0" encoding="ISO-8859-15"?>
<launch>
  <arg name="visualizer_use_zmq" default="false" doc="use zmq to receive scene graphs"/>
  <arg name="visualizer_zmq_ip" default="127.0.0.1" doc="zmq IP to listen on"/>
  <arg name="visualizer_zmq_port" default="8001" doc="zmq port to listen on"/>
  <arg name="visualizer_zmq_url" default="tcp://$(arg visualizer_zmq_ip):$(arg visualizer_zmq_port)" doc="full zmq url"/>

  <arg name="visualizer_frame" default="map"/>
  <arg name="semantic_map_path" default="''"/>
  <arg name="color_mesh_by_label" default="true"/>
  <arg name="gt_regions_filepath" default="''"/>
  <arg name="visualizer_config_path" default="$(find hydra_ros)/config/visualizer_config.yaml"/>
  <arg name="visualizer_plugins_path" default="$(find hydra_ros)/config/visualizer_plugins.yaml"/>

  <arg name="verbosity" default="0"/>
  <arg name="visualizer_debug" default="false"/>
  <arg name="visualizer_roslog_destination" default="log"/>
  <arg name="visualizer_launch_prefix" value="$(eval 'gdb -ex run --args' if arg('visualizer_debug') else '')"/>

  <node pkg="hydra_ros" type="hydra_visualizer_node" name="hydra_dsg_visualizer"
        launch-prefix="$(arg visualizer_launch_prefix)"
        output="$(eval 'screen' if arg('visualizer_debug') else arg('visualizer_roslog_destination'))"
        required="true"
        args="-alsologtostderr -colorlogtostderr -v=$(arg verbosity)">
    <param name="scene_graph_filepath" value=""/>
    <param name="output_path" value=""/>
    <param name="load_graph" value="false"/>
    <param name="use_zmq" value="$(arg visualizer_use_zmq)"/>
    <param name="zmq_url" value="$(arg visualizer_zmq_url)"/>
    <param name="visualizer_frame" value="$(arg visualizer_frame)"/>
    <rosparam file="$(arg visualizer_config_path)" ns="config"/>
    <rosparam file="$(arg visualizer_plugins_path)" subst_value="true"/>
  </node>
</launch>
